api/dataset.go

```diff
@@ -877,14 +918,19 @@ func (api *DatasetAPI) getDimensionOptions(w http.ResponseWriter, r *http.Reques
 	dimension := vars["dimension"]
 
 	logData := log.Data{"dataset_id": datasetID, "edition": editionID, "version": versionID, "dimension": dimension}
+
+	var authorised bool
```

Comment:
This is repeated several times, could be worth creating a function is a wrapper to

```go
authenticate(r)
```

but allows you pass in a reference to logData so the values cane be set in the function